src/snowflake/snowpark/modin/plugin/extensions/base_overrides.py

```diff
@@ -1889,16 +1894,23 @@ def pct_change(
     if limit is lib.no_default:
         limit = None
 
-    if "axis" in kwargs:
-        kwargs["axis"] = self._get_axis_number(kwargs["axis"])
+    kwargs["axis"] = self._get_axis_number(kwargs.get("axis", 0))
 
     # Attempting to match pandas error behavior here
     if not isinstance(periods, int):
         raise TypeError(f"periods must be an int. got {type(periods)} instead")
 
+    if (
+        kwargs["axis"] == 1
+        and any(is_timedelta64_dtype(t) for t in self._get_dtypes())
```

Comment:
There is a `is_timedelta64_dtype` function in QC which is optimized for DatetimeTZDtype. Can you try to use that one?

Reply:
Done.